File: 0142-linked-list-cycle-ii/0142-linked-list-cycle-ii.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode detectCycle(ListNode head) {
       if(head==null) return null;
        
        ListNode curNode=head;
        ArrayList<ListNode> list=new ArrayList<>();
         
        while(curNode!=null){
            if(list.contains(curNode)){
                return curNode;
            }
            list.add(curNode);
            curNode=curNode.next;
        }
        return null; 
    }
}

File: 0142-linked-list-cycle-ii/0142-linked-list-cycle-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def detectCycle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head: return None
        seen=set()
        while head:
            if head not in seen:
                seen.add(head)
            else:
                return head
            head=head.next
        return None
        